Source/WebCore/rendering/RenderMenuList.cpp

```diff
@@ -116,21 +116,36 @@ void RenderMenuList::setInnerRenderer(RenderBlock& innerRenderer)
 
 void RenderMenuList::adjustInnerStyle()
 {
+    auto isHorizontalWritingMode = style().isHorizontalWritingMode();
+
     auto& innerStyle = m_innerBlock->mutableStyle();
     innerStyle.setFlexGrow(1);
     innerStyle.setFlexShrink(1);
     // min-width: 0; is needed for correct shrinking.
-    innerStyle.setMinWidth(Length(0, LengthType::Fixed));
+    if (isHorizontalWritingMode)
+        innerStyle.setMinWidth(Length(0, LengthType::Fixed));
+    else
+        innerStyle.setMinHeight(Length(0, LengthType::Fixed));
     // Use margin:auto instead of align-items:center to get safe centering, i.e.
     // when the content overflows, treat it the same as align-items: flex-start.
     // But we only do that for the cases where html.css would otherwise use center.
     if (style().alignItems().position() == ItemPosition::Center) {
-        innerStyle.setMarginTop(Length());
-        innerStyle.setMarginBottom(Length());
+        if (isHorizontalWritingMode) {
+            innerStyle.setMarginTop(Length());
+            innerStyle.setMarginBottom(Length());
+        } else {
+            innerStyle.setMarginLeft(Length());
+            innerStyle.setMarginRight(Length());
+        }
```

Comment:
same as above, we already have RenderStyle::setMarginStart/End, we are just missing setMarginBefore/After

Reply:
Added.